src/depcache.erl

```diff
@@ -373,16 +387,23 @@ init(Config) ->
                                    [set, named_table, {keypos, 1}, protected, {read_concurrency, true}])
             }
     end,
-    State1 = State#state{
+    State = #state{
+        tables = Tables,
         now=now_sec(),
         serial=0,
         wait_pids=dict:new()
     },
     timer:send_interval(1000, tick),
     spawn_link(?MODULE,
                cleanup,
-               [self(), State1#state.meta_table, State1#state.deps_table, State1#state.data_table, MemoryMaxWords]),
-    {ok, State1}.
+               [#cleanup_state{
+                   pid = self(),
+                   tables = Tables,
+                   name = proplists:get_value(name, Config),
+                   memory_max = MemoryMaxWords,
+                   callback = proplists:get_value(callback, Config)
```

Comment:
My intention was that this can be a general callback (to which other depcache events may be added later). The type of event is discovered by the first element in the tuple: `{eviction, Name}`.